js/src/utils/external.ts

```diff
@@ -21,36 +21,48 @@ export async function sendProcessReq(info: {
     );
     return true;
   }
+
   try {
-    // Spawn a child process to execute a Node.js script
+    // Use node-fetch for making HTTP requests
+    const url = new URL(info.url);
     const child = spawn("node", [
       "-e",
       `
-        const http = require('http');
+        const http = require('${url.protocol === "https:" ? "https" : "http"}');
         const options = {
```

Comment:
Consider adding a request timeout to prevent hanging requests:
```typescript
const options = {
  ...options,
  timeout: 5000, // 5 seconds timeout
};
```
This will ensure requests don't hang indefinitely in case of network issues.